### PR TITLE
Add support for `workspace/codeLens/refresh`

### PR DESCRIPTION
**Pull request by ericdallo**
[SPEC](https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#codeLens_refresh)

I intend to use it with [clojure-lsp](https://clojure-lsp.github.io/clojure-lsp/)

### CODE REVIEW

File: lsp-lens.el [hunk 1]
```diff
@@ -309,6 +309,13 @@ CALLBACK - callback for the lenses."
           (funcall callback lsp-lens--backend-cache lsp--cur-version)
         (lsp-lens--backend-fetch-missing lsp-lens--backend-cache callback lsp--cur-version)))))
 
+(defun lsp--lens-on-refresh (workspace)
+  "Clear lens within all buffers of WORKSPACE, refreshing all workspace buffers."
+  (cl-assert (not (eq nil workspace)))
+  (cl-loop
```
Comment (kiennq) [inline lsp-lens.el:315]: nit: why using `cl-loop` instead of `mapc`?

**Review by kiennq — APPROVED** [review]

File: lsp-lens.el (hunk 1 above)
Comment (ericdallo) [reply lsp-lens.el:315]: I just copied from the same did on semantic-tokens 😅 I don't like cl-loop either

**Review by jcs090218 — APPROVED** [review]

File: lsp-lens.el [hunk 2]
```diff
@@ -309,6 +309,13 @@ CALLBACK - callback for the lenses."
...
+   for ws-buffer in (lsp--workspace-buffers workspace) do
+   (lsp-lens-refresh t ws-buffer)))
```
Comment (yyoncho) [inline lsp-lens.el:317]: can we avoid calling refresh for visible buffers unconditionally, and schedule the refresh once the buffer is visible?

Comment (ericdallo) [reply lsp-lens.el:317]: I think I missed this comment, sure, that seems a good idea, how can we do that exactly?

Comment (yyoncho) [reply lsp-lens.el:317]: we have a similar piece of code for flycheck errors. We refresh only the visible buffer. And we should ensure that non visible buffers will refresh when switching to them.

**Pull request merged**